backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js

```diff
@@ -0,0 +1,42 @@
+import { getDriver } from '../../db/neo4j'
+
+export const description = `
+This migration swaps the value stored in Location.lat with the value
+of Location.lng. This is necessary as the values of lat and lng were
+stored incorrectly. For example Hamburg, Germany, was stored with the
+values lat=10.0 and lng=53.55, which is close to the horn of Africa,
+but it is lat=53.55 and lng=10.0
+`
+
+const swap = async function(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+  try {
+    // Implement your migration here.
+    await transaction.run(`
+        MATCH (l:Location) WHERE NOT(l.lat IS NULL)
+        WITH l.lng AS longitude, l.lat AS latitude, l AS location
+        SET location.lat = longitude, location.lng = latitude
+    `)
+    await transaction.commit()
+    next()
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log(error)
+    await transaction.rollback()
+    // eslint-disable-next-line no-console
+    console.log('rolled back')
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-11T11:50:18Z" title="Tuesday, February 11th 2020, 12:50:18 pm +01:00">Feb 11, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,55 @@
+import { getDriver } from '../../db/neo4j'
+
+export const description =
+  'This migration swaps the value stored in Location.lat with the value of Location.lng.'
+
+export async function up(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    // Implement your migration here.
+    await transaction.run(`
+        MATCH (l:Location) WHERE NOT(l.lat IS NULL)
+        WITH l.lng AS longitude, l.lat AS latitude, l AS Location
+        SET Location.lat = longitude, Location.lng = latitude
+    `)
+    await transaction.commit()
+    next()
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log(error)
+    await transaction.rollback()
+    // eslint-disable-next-line no-console
+    console.log('rolled back')
```


```suggestion
    console.log('rolled back')
    throw new Error(error)
```

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-02-12T13:58:50Z" title="Wednesday, February 12th 2020, 2:58:50 pm +01:00">Feb 12, 2020</time>_

---


Done

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-14T11:33:53Z" title="Friday, February 14th 2020, 12:33:53 pm +01:00">Feb 14, 2020</time>_

---


I like this! could you please update the template as well, so we don't need to give this review feedback every time?